include/vcpkg/base/files.h

```diff
@@ -16,7 +16,13 @@
 #include <experimental/filesystem>
 #endif
 
-namespace fs
+#if VCPKG_USE_STD_FILESYSTEM
+namespace stdfs = std::filesystem;
+#else
+namespace stdfs = std::experimental::filesystem;
+#endif
```

Comment:
I don't like this in the global namespace; one of the points of putting this in `fs::stdfs` is so that we can grep for uses of `std::filesystem`. We should instead put `path` and other stuff that we re-exported into `fs` into `Files` or something.

Reply:
other than that, 
![image](https://user-images.githubusercontent.com/83086508/123459885-b64bcc00-d59b-11eb-9788-ee14c746e521.png)

Reply:
>one of the points of putting this in fs::stdfs is so that we can grep for uses of std::filesystem

I don't see how grepping for fs::stdfs is easier than grepping for stdfs?

>We should instead put path and other stuff that we re-exported into fs into Files or something.

I don't think we should re-export anything because it creates ambiguity between when something should us the stdfs one vs the reexported one, and there is literally no language difference from doing so. (Also note that this removes everything that was merely 'using'd' into stdfs)

Reply:
the point is that currently grepping for `stdfs` brings up a whooooole lot of `path`.

Reply:
>we want to avoid using stdfs in any place that isn't files.cpp.

I do not believe that is a convention we followed:

```
PS D:\vcpkg-tool-master> git status
Not currently on any branch.
nothing to commit, working tree clean
PS D:\vcpkg-tool-master> git diff origin/main
PS D:\vcpkg-tool-master> findstr /snpl /c:"stdfs::" *
include\vcpkg\base\files.h:41:    using stdfs::copy_options;
include\vcpkg\base\files.h:42:    using stdfs::directory_iterator;
include\vcpkg\base\files.h:43:    using stdfs::path;
include\vcpkg\base\files.h:44:    using stdfs::perms;
include\vcpkg\base\files.h:108:    using stdfs::file_type;
include\vcpkg\base\files.h:111:    struct file_status : private stdfs::file_status
include\vcpkg\base\files.h:113:        using stdfs::file_status::file_status;
include\vcpkg\base\files.h:114:        using stdfs::file_status::permissions;
include\vcpkg\base\files.h:115:        using stdfs::file_status::type;
src\vcpkg\base\files.cpp:340:            auto result = follow_symlinks ? fs::stdfs::status(p, ec) : fs::stdfs::symlink_status(p, ec);
src\vcpkg\base\files.cpp:414:            return fs::stdfs::copy_symlink(oldpath, newpath, ec);
src\vcpkg\base\files.cpp:616:        this->copy_file(from, to, fs::stdfs::copy_options::none, ec);
src\vcpkg\base\files.cpp:876:            fs::stdfs::recursive_directory_iterator b(dir, ec), e{};
src\vcpkg\base\files.cpp:891:            fs::stdfs::directory_iterator b(dir, ec), e{};
src\vcpkg\base\files.cpp:927:            fs::stdfs::rename(oldpath, newpath, ec);
src\vcpkg\base\files.cpp:1006:        virtual bool remove(const fs::path& path, std::error_code& ec) override { return fs::stdfs::remove(path, ec); }
src\vcpkg\base\files.cpp:1046:                        for (const auto& entry : fs::stdfs::directory_iterator(current_path))
src\vcpkg\base\files.cpp:1066:                        fs::stdfs::remove(current_path, ec);
src\vcpkg\base\files.cpp:1166:                if (stats.type() == fs::stdfs::file_type::directory)
src\vcpkg\base\files.cpp:1193:        virtual bool is_directory(const fs::path& path) const override { return fs::stdfs::is_directory(path); }
src\vcpkg\base\files.cpp:1194:        virtual bool is_regular_file(const fs::path& path) const override { return fs::stdfs::is_regular_file(path); }
src\vcpkg\base\files.cpp:1195:        virtual bool is_empty(const fs::path& path) const override { return fs::stdfs::is_empty(path); }
src\vcpkg\base\files.cpp:1198:            return fs::stdfs::create_directory(path, ec);
src\vcpkg\base\files.cpp:1202:            return fs::stdfs::create_directories(path, ec);
src\vcpkg\base\files.cpp:1206:            fs::stdfs::create_symlink(to, from, ec);
src\vcpkg\base\files.cpp:1210:            fs::stdfs::create_hard_link(to, from, ec);
src\vcpkg\base\files.cpp:1214:            fs::stdfs::copy(oldpath, newpath, opts);
src\vcpkg\base\files.cpp:1221:            return fs::stdfs::copy_file(oldpath, newpath, opts, ec);
src\vcpkg\base\files.cpp:1284:            return fs::stdfs::absolute(path, ec);
src\vcpkg\base\files.cpp:1288:            return fs::stdfs::system_complete(path, ec);
src\vcpkg\base\files.cpp:1319:        virtual fs::path current_path(std::error_code& ec) const override { return fs::stdfs::current_path(ec); }
src\vcpkg\base\files.cpp:1322:            fs::stdfs::current_path(path, ec);
src\vcpkg\binarycaching.cpp:1105:                const auto status = fs::stdfs::status(path);
src\vcpkg\binarycaching.cpp:1106:                if (!fs::stdfs::exists(status))
src\vcpkg\binarycaching.cpp:1112:                if (!fs::stdfs::is_directory(status))
src\vcpkg\build.cpp:1107:        for (auto& port_file : fs::stdfs::recursive_directory_iterator(port_dir))
src\vcpkg\commands.integrate.cpp:504:        if (fs::stdfs::exists(fish_completions_path))
src\vcpkg\commands.integrate.cpp:513:        fs::stdfs::create_symlink(completion_script_path, fish_completions_path);
src\vcpkg\export.prefab.cpp:28:        fs::stdfs::recursive_directory_iterator it(root);
src\vcpkg\export.prefab.cpp:29:        fs::stdfs::recursive_directory_iterator endit;
src\vcpkg\remove.cpp:77:                        fs::stdfs::permissions(target, fs::perms::owner_all | fs::perms::group_all, ec);
src\vcpkg\vcpkgpaths.cpp:200:                        const auto status = fs::stdfs::status(path);
src\vcpkg\vcpkgpaths.cpp:201:                        if (!fs::stdfs::exists(status))
src\vcpkg\vcpkgpaths.cpp:207:                        if (!fs::stdfs::is_directory(status))
src\vcpkg-test\util.cpp:188:            fs::stdfs::create_symlink(targetp, filep, ec);
PS D:\vcpkg-tool-master>
```

Reply:
those are, frankly, all bugs.

Reply:
After some in person discussion this was resolved as:
1. Delete `vcpkg::files`
2. `namespace vcpkg { using stdfs::path; }`
3. Remove `stdfs` from all the places to say path.